### PR TITLE
fix: modern Node

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Review by vweevers — CHANGES_REQUESTED** [review]
Please keep the `shutdown`, `onlistening` and `onopen` functions, they make the code (as well as the diff of this PR) more readable. And where there's no `this` to worry about, let's also keep `function` in favor of arrow functions.

**Comment by ronag** [comment]
Done

File: index.js [hunk 1]
```diff
@@ -1,57 +1,79 @@
...
-          if (connected) return tryConnect()
-          return setTimeout(tryConnect, 100)
+          // TODO: This can cause an invisible retry loop that never completes
+          // and leads to memory leaks.
+          // TODO: What errors should be retried?
+          if (connected) {
+            tryConnect()
+          } else {
+            setTimeout(tryConnect, 100)
+          }
```
Comment (ronag) [inline index.js:52]: This looks like this will constantly retry for all non masters taking quite a lot of cpu.

File: index.js [hunk 2]
```diff
@@ -1,57 +1,79 @@
 'use strict'
 
-var level = require('level')
-var has = require('has')
```
Comment (vweevers) [inline index.js:4]: Are we still using `has` elsewhere, or can it be removed from `package.json`?

File: index.js [hunk 3]
```diff
@@ -1,57 +1,79 @@
...
-var path = require('path')
-var multileveldown = require('multileveldown')
-
-module.exports = function (dir, opts) {
-  if (!opts) opts = {}
-  if (!has(opts, 'retry')) opts.retry = true
-
-  var sockPath = process.platform === 'win32'
+const level = require('level')
+const { pipeline: pump } = require('stream')
```
Comment (vweevers) [inline index.js:4]: In modules it's preferred to use `require('readable-stream')` to get a particular version. Or well, it used to be preferred, at a time where `stream` could mean streams1 or streams2. Still, considering the complexity of streams I'm leaning towards sticking with `require('readable-stream')` as a stable base. What do you think?

File: index.js (hunk 2 above)
Comment (ronag) [reply index.js:4]: I can make sure and then remove

File: index.js (hunk 1 above)
Comment (vweevers) [reply index.js:52]: Let's discuss in a separate issue or PR; if a fix is needed it would be nice to also add a test for it.

File: index.js (hunk 3 above)
Comment (ronag) [reply index.js:4]: We've fixed so many weird edge cases with streams in node 14 so I would really like to have those when possible... I'd be more inclined to use readable-stream when v4 is release.

File: index.js (hunk 1 above)
Comment (ronag) [reply index.js:52]: Sure. Lets leave the TODO's though. I'd be happy to discuss all of them.

File: index.js (hunk 3 above)
Comment (vweevers) [reply index.js:4]: If we can't have those fixes for all the node versions that we support here, it's not worth it IMO. It's just as important that everyone has the same code base. When someone opens a bug report, I don't want to dive into multiple versions of `stream`.

Comment (ronag) [reply index.js:4]: Fair enough.

**Review by vweevers — CHANGES_REQUESTED** [review]
Thanks, looks good! One last nit: remove the lockfile. Lockfiles are good for apps, not so much for modules. It doesn't apply when one does `npm i level-party` so as module authors we shouldn't lock either.

**Comment by ronag** [comment]
Removed

**Review by vweevers — APPROVED** [review]

**Comment by vweevers** [comment]
5.0.0
